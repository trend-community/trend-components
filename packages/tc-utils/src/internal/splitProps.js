/**
 * @param {object} props
 * @param {array} keys
 * @return {Array<object, object} - An array of two objects (picked, omitted).
 */

export default function splitProps(props, keys) {
  const propsKeys = Object.keys(props);
  const picked = {};
  const omitted = {};

  propsKeys.forEach(key => {
    if (keys.indexOf(key) >= 0) {
      picked[key] = props[key];
    }
    else {
      omitted[key] = props[key];
    }
  });

  return [picked, omitted];
}
