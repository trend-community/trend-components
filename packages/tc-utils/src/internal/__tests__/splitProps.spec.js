import splitProps from '../splitProps';

test('splitProps', () => {
  const props = { foo: 'bar', hello: 'world', fee: 'fi-fo' }
  const result = splitProps(props, ['foo']);

  expect(result).toMatchSnapshot();
});
