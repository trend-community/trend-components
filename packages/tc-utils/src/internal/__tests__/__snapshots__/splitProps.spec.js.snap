// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`splitProps 1`] = `
Array [
  Object {
    "foo": "bar",
  },
  Object {
    "fee": "fi-fo",
    "hello": "world",
  },
]
`;
